tests/array.rs

```diff
@@ -807,6 +816,49 @@ fn owned_array_with_stride() {
     assert_eq!(a.strides(), &[1, 4, 2]);
 }
 
+#[test]
+fn owned_array_discontiguous() {
+    use ::std::iter::repeat;
+    let v: Vec<_> = (0..12).flat_map(|x| repeat(x).take(2)).collect();
+    let dim = (3, 2, 2);
+    let strides = (8, 4, 2);
+
+    let a = Array::from_shape_vec(dim.strides(strides), v).unwrap();
+    assert_eq!(a.strides(), &[8, 4, 2]);
+    println!("{:?}", a.iter().cloned().collect::<Vec<_>>());
+    itertools::assert_equal(a.iter().cloned(), 0..12);
+}
+
+#[test]
+fn owned_array_discontiguous_drop() {
+    use ::std::rc::Rc;
+    use ::std::cell::RefCell;
+    use ::std::collections::BTreeSet;
+
+    struct InsertOnDrop<T: Ord>(Rc<RefCell<BTreeSet<T>>>, Option<T>);
+    impl<T: Ord> InsertOnDrop<T> {
+        fn new(set: Rc<RefCell<BTreeSet<T>>>, value: T) -> Self {
```

Comment:
Will you please remove this `new` method and just construct `InsertOnDrop` manually with `InsertOnDrop(set.clone(), Some(x))`? (`new` is used in only one place, and it would be nice to make the test more compact. I see the advantage of hiding the implementation detail that `value` should start as `Some`, but since the use is so close to the implementation, I prefer conciseness over encapsulation in this case.)

Reply:
No problem, that was an artefact of copypasta from another project where it was used more heavily